ubuntu2004/Dockerfile

```diff
@@ -92,9 +92,17 @@ RUN mkdir src \
   && cd .. \
   && rm -rf src
 
+# nlohmann's JSON
+RUN mkdir src \
+  && ${GET} https://github.com/nlohmann/json/archive/refs/tags/v3.10.2.tar.gz \
+    | ${UNPACK_TO_SRC} \
+  && cmake -B build -S src -GNinja \
+  && cmake --build build -- install \
+  && rm -rf build src
```

Comment:
Having a separate installation of nlohmann_json is not a bad thing. But if we don't desperately need such a separate installation, you could've just as well added

```
-Dbuiltin_nlohmann_json:BOOL=TRUE
```

to the ROOT configuration command, and it would've built a viable version of this external itself.

I think it's purely a maintainability question at this point, since your solution seems to work as well.

Reply:
I used the same strategy that we use for xxhash, which is also available from the ROOT source, but is installed as a separate layer in the Docker image. In the end, as far as I am concerned, projects building their own dependencies is a fragile antipattern and I think avoiding it is a lot more robust, even if just for a CI Docker image. That said if anyone feels strongly about it we can use the "builtin" approach. :slightly_smiling_face: